external/redis/redis.go

```diff
@@ -0,0 +1,48 @@
+package redis
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+)
+
+// var redisClient *redis.Client
+
+type redisClient struct {
+	client *redis.Client
+}
+
+//go:generate mockgen -source=redis.go -destination=mock/redis.go -package=mock
+type RedisInterface interface {
+	GetContext() context.Context
+	GetStringKey(ctx context.Context, key string) (string, error)
+	Set(ctx context.Context, key string, value interface{}, expiration time.Duration) error
+}
+
+func Connection() *redisClient {
+	redisServer := os.Getenv("REDIS_SERVER")
+	// Initialize Redis connection
+	client := redis.NewClient(&redis.Options{
+		Addr:     redisServer, // Your Redis server address
+		Password: "",          // No password
```

Comment:
I think it is better not to hardcoded the password and database mas even if the value is default, we could store them in the .env like REDIS_SERVER

Reply:
updated mas, thank you